### ISSUE TITLE
Crash when specifying the wrong tag-template

### ISSUE THREAD

**Issue by Alexander-Wilms** [flathub/flatpak-external-data-checker#122]
The tool crashes when trying to get a tagged commit that doesn't exist. I used $version instead of v$version for qtconnectivity's tag-template and got the following crash:

```
+ 2021-02-12 16:48:28,763    INFO src.checker: [7/13] checking qtconnectivity
Traceback (most recent call last):
  File "/app/flatpak-external-data-checker", line 30, in <module>
    main()
  File "/app/src/main.py", line 301, in main
    if run_with_args(parse_cli_args()) > 0:
  File "/app/src/main.py", line 277, in run_with_args
    manifest_checker.check(filter_type)
  File "/app/src/checker.py", line 183, in check
    checker.check(data)
  File "/app/src/checkers/anityachecker.py", line 36, in check
    return self._check_git(external_data, latest_version)
  File "/app/src/checkers/anityachecker.py", line 57, in _check_git
    timestamp=None,
  File "/app/src/lib/externaldata.py", line 258, in fetch_remote
    got_commit = self._get_tagged_commit(refs, self.tag)
  File "/app/src/lib/externaldata.py", line 246, in _get_tagged_commit
    raise KeyError(f"refs/tags/{tag}")
KeyError: 'refs/tags/5.15.2'
```

<details>

```
app-id: org.subsurface_divelog.Subsurface
sdk: org.kde.Sdk
runtime: org.kde.Platform
runtime-version: '5.15'
base: io.qt.qtwebkit.BaseApp
base-version: '5.15'
command: subsurface
rename-icon: subsurface-icon
rename-appdata-file: subsurface.appdata.xml
rename-desktop-file: subsurface.desktop
finish-args:
- --socket=wayland
- --socket=fallback-x11
- --share=ipc
- --share=network
- --device=all # required to talk to dive computers
- --system-talk-name=org.bluez # required to talk to dive computers
- --filesystem=xdg-config/kdeglobals:ro # gives application access to kdeglobals
- --talk-name=com.canonical.AppMenu.Registrar # gives application access to dbus menu
cleanup:
- /include
- /lib/pkgconfig
- /lib64/pkgconfig
- /lib/cmake
- /share/doc
- /share/gtk-doc
- /share/man
- /share/pkgconfig
- /src
- "*.a"
- "*.la"
- "*.cmake"
modules:
- shared-modules/libusb/libusb.json

- name: libical
  buildsystem: cmake-ninja
  config-opts:
  - "-DCMAKE_INSTALL_LIBDIR=lib"
  - "-DBUILD_SHARED_LIBS:BOOL=ON"
  - "-DICAL_GLIB=true"
  - "-DGOBJECT_INTROSPECTION=true"
  - "-DICAL_GLIB_VAPI=true"
  - "-DICAL_BUILD_DOCS=false"
  - -DCMAKE_BUILD_TYPE=RelWithDebInfo
  - -DICAL_GLIB_VAPI=False
  sources:
  - type: archive
    url: https://github.com/libical/libical/archive/v3.0.9.tar.gz
    sha256: bd26d98b7fcb2eb0cd5461747bbb02024ebe38e293ca53a7dfdcb2505265a728
    x-checker-data:
      type: anitya
      project-id: 1637
      url-template: https://github.com/libical/libical/archive/v$version.tar.gz
  cleanup:
    - /lib/cmake

- name: libssh2
  buildsystem: cmake
  config-opts:
  - -DCMAKE_BUILD_TYPE=RelWithDebInfo
  - -DCMAKE_INSTALL_LIBDIR=/app/lib
  - -DCMAKE_C_FLAGS=-fPIC
  sources:
  - type: archive
    url: https://github.com/libssh2/libssh2/archive/libssh2-1.9.0.tar.gz
    sha256: db257d41754b16518c3bd0c3b8dd08940ccbee5fc44cef7bfb052d930bcbbda3
    x-checker-data:
      type: anitya
      project-id: 1730
      url-template: https://github.com/libssh2/libssh2/archive/libssh2-$version.tar.gz

- name: libzip
  buildsystem: cmake-ninja
  config-opts:
  - -DCMAKE_BUILD_TYPE=RelWithDebInfo
  sources:
  - type: archive
    url: https://github.com/nih-at/libzip/archive/v1.7.3.tar.gz
    sha256: c0a086901dfc8bda0fb8695f2d3c8050ed140b7899536b9348bcc72b47b2f307
    x-checker-data:
      type: anitya
      project-id: 10649
      url-template: https://github.com/nih-at/libzip/archive/v$version.tar.gz

- name: grantlee
  buildsystem: cmake-ninja
  config-opts:
  - -DCMAKE_BUILD_TYPE=RelWithDebInfo
  sources:
  - type: archive
    url: https://github.com/steveire/grantlee/archive/v5.2.0.tar.gz
    sha256: 139acee5746b957bdf1327ec0d97c604d4c0b9be42aec5d584297cb5ed6a990a
    x-checker-data:
      type: anitya
      project-id: 21448
      url-template: https://github.com/steveire/grantlee/archive/v$version.tar.gz
        
- name: libgit2
  buildsystem: cmake-ninja
  config-opts:
  - -DCMAKE_BUILD_TYPE=RelWithDebInfo
  sources:
  - type: archive
    url: https://github.com/libgit2/libgit2/archive/v1.1.0.tar.gz
    sha256: 41a6d5d740fd608674c7db8685685f45535323e73e784062cf000a633d420d1e
    x-checker-data:
      type: anitya
      project-id: 1627
      url-template: https://github.com/libgit2/libgit2/archive/v$version.tar.gz

- name: bluez
  config-opts:
  - "--disable-datafiles"
  - "--disable-systemd"
  - "--enable-experimental"
  - "--enable-library"
  - "--disable-client"
  - "--disable-mesh"
  - "--disable-tools"
  - "--disable-monitor"
  - "--disable-udev"
  - "--prefix=/app"
  - "--sysconfdir=/app/etc"
  sources:
  - type: archive
    url: http://www.kernel.org/pub/linux/bluetooth/bluez-5.54.tar.xz
    sha256: 68cdab9e63e8832b130d5979dc8c96fdb087b31278f342874d992af3e56656dc
    x-checker-data:
      type: anitya
      project-id: 10029
      url-template: http://www.kernel.org/pub/linux/bluetooth/bluez-$version.tar.xz
      
- name: qtconnectivity
  buildsystem: simple
  cleanup-platform:
  - /bin
  - /mkspecs
  sources:
  - type: git # build fails when using the archive
    url: https://github.com/qt/qtconnectivity
    branch: 5.14.2
    commit: b445ea4ff203fbcb5011adae0236dd028f11bc89
    x-checker-data:
      type: anitya
      project-id: 153467
      tag-template: $version
  build-commands:
  - qmake
  - make -j $FLATPAK_BUILDER_N_JOBS
  - cp -r -n bin /app
  - cp -r -n include /app
  - cp -r -n lib /app
  - mkdir -p /app/src/bluetooth
  - cp -r src/bluetooth /app/src/

- name: googlemaps
  buildsystem: simple
  sources:
  - type: git
    url: https://github.com/Subsurface-divelog/googlemaps.git
  build-commands:
  - mkdir build
  - cd build &&
    qmake PREFIX=/app ../googlemaps.pro &&
    sed -i 's|(INSTALL_ROOT)/usr|//app|g' Makefile &&
    make -j $FLATPAK_BUILDER_N_JOBS &&
    make install

- name: hidapi
  sources:
  - type: archive
    url: https://github.com/libusb/hidapi/archive/hidapi-0.10.1.tar.gz
    sha256: f71dd8a1f46979c17ee521bc2117573872bbf040f8a4750e492271fc141f2644
    x-checker-data:
      type: anitya
      project-id: 5594
      url-template: https://github.com/libusb/hidapi/archive/hidapi-$version.tar.gz
      
- name: libconfuse # build dependency of libftdi
  sources:
  - type: archive
    url: https://github.com/libconfuse/libconfuse/archive/v3.3.tar.gz
    sha256: cb90c06f2dbec971792af576d5b9a382fb3c4ca2b1deea55ea262b403f4e641e
    x-checker-data:
      type: anitya
      project-id: 1581
      url-template: https://github.com/libconfuse/libconfuse/archive/v$version.tar.gz

- name: libftdi
  buildsystem: cmake-ninja
  config-opts:
  - -DLIB_SUFFIX=
  - -Wno-dev
  sources:
  - type: archive
    url: https://www.intra2net.com/en/developer/libftdi/download/libftdi1-1.5.tar.bz2
    sha256: 7c7091e9c86196148bd41177b4590dccb1510bfe6cea5bf7407ff194482eb049
    x-checker-data:
      type: anitya
      project-id: 14622
      url-template: https://www.intra2net.com/en/developer/libftdi/download/libftdi1-$version.tar.bz2
      
- name: libdc
  sources:
  - type: archive
    url: https://github.com/subsurface/libdc/archive/v4.9.10.tar.gz
    sha256: 2bf53c2fa23ae31ce6df803d9668a7db22cc699b7fac330d6b303547c93d65a6
    x-checker-data:
      type: anitya
      project-id: 153474
      url-template: https://github.com/subsurface/libdc/archive/v$version.tar.gz
  - type: shell
    commands:
    - autoreconf --install

- name: subsurface
  buildsystem: cmake
  builddir: true
  config-opts:
  - -DCMAKE_BUILD_TYPE=RelWithDebInfo
  - -DFTDISUPPORT=ON
  - -DQt5WebKitWidgets_DIR=/app/lib/cmake/Qt5WebKitWidgets
  sources:
  - type: git # use git in order to make HandleVersionGeneration.cmake work
    url: https://github.com/Subsurface-divelog/subsurface
    branch: v4.9.10
    commit: f4ef9565a7dca695812870a87e5c25a3605ac5bb
    x-checker-data:
      type: anitya
      project-id: 4903
      tag-template: v$version
  - type: shell
    commands:
    - sed -e "s|setDesktopFileName(\"subsurface\")|setDesktopFileName(\"org.subsurface_divelog.Subsurface\")|" -i core/qt-init.cpp
    - sed -e "s|<id>subsurface</id>|<id>org.subsurface_divelog.Subsurface</id>|" -i appdata/subsurface.appdata.xml.in
```
</details>

**Comment by gasinvein** [comment]
It's kinda expected behavior. We fail gracefully only in catch-all checkers that are applied without maintainer directions. Otherwise - raise exception if something goes wrong.

I guess we need some policy how to handle errors in case of
- Invalid input (e.g. malformed checker data)
- Remote error (e.g. missing git tag or url with http 404)

**Comment by gasinvein** [comment]
Fixed in #207
